### DISCUSSION TITLE
One way binding to AlpineJS

### DISCUSSION THREAD

**Discussion by Mushr0000m** [livewire/livewire#5487] [1. Help]
I use Liverwire and Alpine on multiple projects but always have a hard time passing data from Livewire to Alpine. Not the very usefull `@entangle()` that does two-way binding but just in one way from Livewire to Alpine.

The code below doesn't work because the alpine component `init` method would be called each time livewire re-render : 
```
<div x-data="map(@js($this->track->markers))"></div>
```

This doesn't work either because the Alpine component doesn't see the update on the `markers` : 
```
<div x-data="map(@js($this->track->markers))" wire:ignore></div>
```

And `@entangle` doesn't work in my case because it's a two way binding and I need to do some things on the JS side before sending the data back to the livewire component. Also as you can see I'm accessing `$this->track->markers` for the relationship `markers` on `track` and `@entangle('track.markers')` is not supported from what I know of.

I must have missed something here, hope you can help !

**Comment by joshhanley** [comment]
@Mushr0000m generally if you need to access Livewire data from Alpine, you should use `$wire` if `@entangle` isn't suitable. Updating the data echoed in blade with `@js` won't work as once Alpine is initialised it wouldn't pick up the changes.

But entangle should work with nested data. If they are Eloquent models, make sure you have them in your `$rules` array so the data is sent to the front end where Alpine can see it.

Hope this helps!
